### PR TITLE
HTML Reporter: Handle URL params named like Object.prototype properties

### PR DESCRIPTION
**Pull request by gibson042**
Fixes gh-1029

### CODE REVIEW

File: reporter/urlparams.js [hunk 1]
```diff
@@ -85,7 +86,7 @@ function getUrlParams() {
 			// Allow just a key to turn on a flag, e.g., test.html?noglobals
 			value = param.length === 1 ||
 				decodeQueryParam( param.slice( 1 ).join( "=" ) ) ;
-			if ( urlParams[ name ] ) {
+			if ( hasOwn.call( urlParams, name ) ) {
```
Comment (trentmwillis) [inline reporter/urlparams.js]: I think another possible approach would be to simply initialize the object like:

```js
var urlParams = Object.create(null);
```

Then we don't have to worry about any property collisions as there are none to inherit. Thoughts?

Comment (gibson042) [inline reporter/urlparams.js]: It's an old compatibility habit. But since we're now on IE9+, I'll make the change.

File: reporter/urlparams.js [hunk 2]
```diff
@@ -85,7 +85,7 @@ function getUrlParams() {
 			// Allow just a key to turn on a flag, e.g., test.html?noglobals
 			value = param.length === 1 ||
 				decodeQueryParam( param.slice( 1 ).join( "=" ) ) ;
-			if ( urlParams[ name ] ) {
+			if ( name in urlParams ) {
```
Comment (leobalter) [inline reporter/urlparams.js]: We can also use `Object#hasOwnProperty` here, I'm not a big fan of the in operator and we don't need to search the property in the urlParams prototype, as it is null.

If you prefer to keep the in operator, I'm ok with it, there's nothing wrong with it.

File: reporter/urlparams.js (hunk 1 above)
Comment (leobalter) [inline reporter/urlparams.js]: HA! Now I just saw this comment...

Well, the Object.create(null) is great, but we can stick with the hasOwn :)

Comment (leobalter) [inline reporter/urlparams.js]: btw, spec wise, hasOwn will take less steps until it returns a value than the in operator. Performance wise I can't tell which is faster.

**Pull request closed without merging**